docs/UserGuide.md

```diff
@@ -246,16 +247,43 @@ Restrictions:
 * The role argument must be an integer, followed by a space, and then the name of role required.
 * The budget argument must be a number in 2 decimal places.
+* The maximum number of volunteers argument must be a non-negative integer.
 
 <box type="tip" seamless>
 
-**Tip:** If the end date and time is not specified, iVolunteer will automatically set the end date and time to **exactly 3 hours** after the start date and time.
+**Tips:** 
+- If the end date and time is not specified, iVolunteer will automatically set the end date and time to **exactly 3 hours** after the start date and time.
+- If the maximum number of volunteers argument is not specified, iVolunteer will not set a limit to the number of volunteers within the event.
+  - Alternatively, if the maximum number of volunteers is set to 0 (e.g. `ecreate ... vs/0`), iVolunteer will remove a limit to the number of volunteer within the event.
 </box>
 
 Examples:
 * `ecreate n/clean beach r/10 cleaner sd/30/11/2023 1200 ed/30/11/2023 1800 l/east coast park dsc/help clean east coast park m/10 pairs of gloves m/10 trash bags b/50.00` 
   * Creates an event with name `clean beach`, roles needed `10 cleaner`, event date from `30th November 2023, 12pm` to `30th November 2023, 6pm`, location `east coast park`, description `help clean east coast park`, materials needed `10 pairs of gloves` and `10 trash bags` and budget `$50.00`
 
+<box type="info" seamless>
+
+**How do you fill up the *role* and *material* information within the UI?**
+
+![Role and material in UI](images/user-guide/event-role-material.png)
+
+- To add a **volunteer** with a certain role, you can use the [`eaddv` command](#adding-a-volunteer-into-an-event-eaddv) as follows:
+    1. Ensure that the volunteer's skill matches the role of the event you want to fill up (both role and skills are case-sensitive).
+        - In the example image above, volunteers with skills `chef` or `cooking` can be added. If not, you can use the [`eedit` command](#edit-the-details-of-an-event-eedit)
+          (e.g. `eedit VOLUNTEER_INDEX s/cooking s/other_skill`) to change the volunteer's skills to match the role.
+    2. Use the `eaddv eid/EVENT_INDEX vid/VOLUNTEER_INDEX` command to add the volunteer to the event. For example, if the volunteer with skill `cooking` you want to add is in
+       index `3` of the displayed volunteer list, and the event with role `cooking` required is in index `4` of the displayed event list, then you can use `eaddv vid/3 eid/4`
+       to add that volunteer into the event.
+    3. You should notice that the `cooking` role would be **incremented by 1** in the UI. In the example above, since `1 / 2 cooking` skills has been filled up, the addition of the new volunteer
+       will cause the role `cooking` to be updated to `2 / 2 cooking`. The role will also turn green since there are already 2 volunteers with role `cooking`.
```

Comment:
maybe can add photos along the way for them to refer to?

Reply:
before picture is alr there, 2 after pictures will be added (i wanted to add but thanks for the confirmation)

Reply:
yessir